src/stream/src/executor/global_simple_agg.rs

```diff
@@ -290,16 +289,17 @@ impl<S: StateStore> GlobalSimpleAggExecutor<S> {
                     .await?;
                 }
                 Message::Barrier(barrier) => {
-                    let next_epoch = barrier.epoch.curr;
-                    if let Some(chunk) =
-                        Self::flush_data(&info.schema, &mut states, epoch, &mut state_tables)
-                            .await?
+                    if let Some(chunk) = Self::flush_data(
+                        &info.schema,
+                        &mut states,
+                        barrier.epoch,
```

Comment:
Open for discussions:
I am thinking whether we can also avoid passing epoch in `Executor::flush_data` and `StateTable::commit` after we have stored current epoch in StateTable. Here are my thoughts:
- On `StateTable::commit`, use `epoch.curr` to write batch to state store.
- `StateTable` always uses its `epoch.curr` to read from state store.
- After processing `Message::Barrier` and yielding the Barrier, update the epoch in `StateTable`.

In this case, we don't need to pass epoch inside executor execution. Also, only `epoch.curr` is used and `epoch.prev` is only for validation.

Reply:
cc @st1page @wcy-fdu

Reply:
LGTM! I consider this PR is a part of the streaming executors and I think they do not need to be aware of the epoch by themselves :+1:

Reply:
I saw that lookup executor has a `use_current_epoch` flag. Do you by any chance know when it will be false? If it is set to false, we need to use prev epoch to read from storage, which breaks the assumption that streaming executor also read latest data. My proposal won't work if we need to handle this case.

Reply:
Discussed offline. Lookup executor indeed breaks the "read from latest" assumption because it needs to merge S3 state of prev_epoch and delta buffer state of curr_epoch. Therefore, we need to handle this special case if adopting my proposal.

In other words, lookup executor needs to be aware of epoch in its implementation but all other streaming executors don't need to be aware of epoch.

Reply:
I have a question: when or in which case will `use_current_epoch == false`？